src/canvas/SelectableCanvas.ts

```diff
@@ -854,38 +839,67 @@ export class SelectableCanvas<EventSpec extends CanvasEvents = CanvasEvents>
   }
 
   /**
-   * Returns pointer coordinates without the effect of the viewport
-   * @param {Object} pointer with "x" and "y" number values in canvas HTML coordinates
-   * @return {Object} object with "x" and "y" number values in fabricCanvas coordinates
+   * @returns point existing in the same plane as the {@link HTMLCanvasElement},
+   * `(0, 0)` being the top left corner of the {@link HTMLCanvasElement}.
+   * This means that changes to the {@link viewportTransform} do not change the values of the point
+   * and it remains unchanged from the viewer's perspective.
+   *
+   * @example
+   * const pointInCanvasPlane = sendPointToPlane(
+   *  this.getHTMLPointFromEvent(e),
+   *  undefined,
+   *  canvas.viewportTransform
+   * );
+   *
    */
-  restorePointerVpt(pointer: Point): Point {
-    return pointer.transform(invertTransform(this.viewportTransform));
+  getHTMLPointFromEvent(e: TPointerEvent) {
+    if (this._pointer) {
+      return this._pointer;
+    }
+    return this.getPointer(e, true);
+  }
+
+  /**
+   * @returns point existing in the canvas plane (the same plane as the plane {@link FabricObject#getCenterPoint} exists in),
+   * relative to instance's `(0, 0)`.
+   * This means that changes to the {@link viewportTransform} do not change the values of the point,
+   * however, from the viewer's perspective the point is changed.
+   *
+   * @example
+   * const pointInHTMLPlane = sendPointToPlane(
+   *  this.getCanvasPointFromEvent(e),
+   *  canvas.viewportTransform
+   * );
+   *
+   */
+  getCanvasPointFromEvent(e: TPointerEvent) {
+    if (this._absolutePointer) {
+      return this._absolutePointer;
```

Comment:
Risky business IMO, the method sounds a pure calculation but this could return some stale state

Reply:
Also this is also handled internally in `getPointer`

Reply:
I forgot to remove from there
but yes, caching here seems...

Reply:
however all events run or at least should run `_cacheTransformEventData` and `_resetTransformEventData`
I agree it is far from perfect

Reply:
getting the pointer traversing the dom can be expensive, that is why caching was added.
An event chain can do that multiple times, the risky businness of not having stale cache is on us. The events are syncronous and they create and delete the cache, a user could save an event object from yesterday, and use it today on an event handler and yes get a wrong result. We should document that getPointer inside an event handler is meant to return you the pointer for that event and not a previous mouse event.